projects/ADuCM3029_demo_cn0428_cn0429/src/ADuCM3029_demo_cn0428_cn0429.cpp

```diff
@@ -0,0 +1,1307 @@
+
+		for(loop = 1; loop < string_len;
+		    loop++) {                                 // Loop through the match string and start at 1 as already found first character
+			read_char = GetChar();
+			if(read_char == string_to_search[loop]) {
+				match_success += 1;
+			}
+		}
+		if(match_success ==
+		    string_len) {                                          // have all chars matched ?
+			DeleteChar(string_len);												// delete command from the ring buffer
+			return Success;
+		} else {
+			Empty_Ring();
+			return Failure;
+		}
+	}
+}
+
+/*!
+ * @brief      Initializes gas sensor and slave I2C address
+ *
+ * @details    Called for each site, so the I2C address is set at run-time based on the site
+ * 				the board is plugged into. Slave has a function that sets its I2C address to
+ *				new_sensor_address only if the /SS GPIO is set low. This function sets the
+ *				gpios and then calls that function.
+ */
+SENSOR_RESULT SensorInitwithI2CAddr(uint8_t new_sensor_address, uint8_t site)
+{
+	uint8_t dataPayload	[64];
+	uint8_t i = 0;
+
+	//	Set target site /CS low and all other sites high
+	for(i = 0; i<4; i++) {
+		if(site == i + 1) {
+			eGpioResult = adi_gpio_SetLow(CS_Port[i],
+						      CS_Pin[i]);				// Set Sensor CS Low if correct site
+		} else {
+			eGpioResult = adi_gpio_SetHigh(CS_Port[i],
+						       CS_Pin[i]);				// Otherwise set Sensor CS High
+		}
+		delay_ms(50);
+	}
+
+	eSensorResult = pGasSensor->SetI2CAddr(new_sensor_address);
+	if (eSensorResult == SENSOR_ERROR_NONE) {
+		//DEBUG_MESSAGE("Sensor: 0x%x initialized successfully!\r\n", sensor_address);
+		sprintf((char*)&dataPayload,
+			" Gas Sensor in site %d initialized successfully with address 0x%02X!%s", site,
+			new_sensor_address, _EOS);
+	} else if (eSensorResult ==
+		   SENSOR_ERROR_PH) { //Used to identify that a water quality probe was found
+		//DEBUG_MESSAGE("Sensor: 0x%x initialization error!\r\n", sensor_address);
+		sprintf((char*)&dataPayload,
+			" Water Quality Sensor in site %d initialized successfully. Address 0x%02X!%s",
+			site, new_sensor_address, _EOS);
+	} else {
+		//DEBUG_MESSAGE("Sensor: 0x%x initialization error!\r\n", sensor_address);
+		sprintf((char*)&dataPayload,
+			" Sensor in site %d: 0x%02X initialization error. Address 0x%02X!%s", site,
+			(unsigned int) eSensorResult, new_sensor_address, _EOS);
+	}
+	UART_TX((const char*)dataPayload);
+	pGasSensor->close();
+
+	return eSensorResult;
+}
+
+/*!
+ * @brief      Read sensor config
+ *
+ * @details    Reads sensor config of the sensor with address specified in input parameter
+ */
+SENSOR_RESULT GetSensorCfg(uint8_t sensor_address)
+{
+	uint8_t pBuff[8] = {0};
+	int16_t sigVal = 0;
+	uint16_t unsigVal = 0;
+	uint32_t unsiglVal = 0;
+
+	eSensorResult = pGasSensor->openWithAddr(sensor_address);
+	if(eSensorResult != SENSOR_ERROR_NONE) return eSensorResult;
+	delay_ms(5);
+
+	if (sensor_address == 0x0A) sprintf ((char*)&TXbuff,
```

Comment:
this also looks repetitive; it could be done with an offset
```
offset = (sensor_address - ADI_CFG_I2C_SENSOR1_ADDR + 1);
sprintf ((char*)&TXbuff,  "Config of sensor on site %d:%s", ( _EOS);
```